backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -1458,29 +1466,33 @@ export class ExperimentService {
   }
 
   private async addFactorialDataInDB(
-    partitions: DecisionPoint[],
+    factors: any[],
     conditions: ExperimentCondition[],
+    experimentDoc: Experiment,
     transactionalEntityManager: EntityManager,
     logger: UpgradeLogger
   ) {
-    // creating factorDoc
-    const allFactors: Factor[] = [];
     const allLevels: Level[] = [];
     const allLevelCombinationElements: LevelCombinationElement[] = [];
 
-    partitions.forEach((partition) => {
-      partition.factors.forEach((factor) => {
+    // creating factors docs
+    const allFactors =
+      factors &&
+      factors.length > 0 &&
+      factors.map((factor) => {
```

Comment:
this `map` returns the factor to nowhere, but I think `forEach` is what was meant?

Reply:
Okay, you are right. Thanks

Reply:
We need a map so that the value from different factors combined to allFactors and also using forEach sometimes throw some null error in further code.

Reply:
my bad, I misread the code 👍